### ISSUE TITLE
Proposals disappearing right away (the second time searchTF is used)

### ISSUE THREAD

**Issue by frederic-adda** [apasccon/SearchTextField#27]
The first time  I use the searchTextField, the proposals are displayed nicely
![simulator screen shot 6 mars 2017 a 18 04 39](https://cloud.githubusercontent.com/assets/17590846/23620545/993b2c90-0297-11e7-8ed7-3b96fb13605e.png)

But the second time I display the searchTF, the proposals appear, and then disappear right away before I have time to select an entry.
![simulator screen shot 6 mars 2017 a 18 05 08](https://cloud.githubusercontent.com/assets/17590846/23620567/b5eebdd4-0297-11e7-9d2e-b0d1932d0b8c.png)

Maybe an issue with `userStoppedTypingHandler` ?
This didn't happen in release 1.0.4, then I upgraded to 1.0.9 and the issue appeared.

Thanks !

**Comment by apasccon** [comment]
Hey!, I'm not able to reproduce the issue. Probably you can help me identifying better in which situation it happens. I can see that you are using it in a kind of pop up component. Which one is it?. If you can give me more information I'll appreciate it. Thanks!.

**Comment by frederic-adda** [comment]
Yes, I use it in a SCLAlertView, which is basically a kind of nice alert view. It worked fine in 1.0.4. I downgraded to 1.0.4 to check. In 1.0.7 it's not working anymore, but I could try and downgrade version by version to identify a bit more precisely.

**Comment by frederic-adda** [comment]
Hi again
I checked again, and I'm not able to reproduce the issue. 
Sorry for the inconvenience, it probably didn't have anything to do with the SearchTextField ;-)

Thanks
Fred

**Comment by frederic-adda** [comment]
Sorry, I have to reopen the issue !
It appeared in 1.0.6 -> up to 1.0.5, working fine, but from 1.0.6 on, the second time I use `searchTextField.filterItems(items)`, the results appear very quickly and then disappear.
I don't know if someone else experienced this issue.

Since I didn't change anything in my code, I just changed the podfile, going from 1.0.5 to 1.0.6, I guess it has to do with the SearchTextField.

Thanks !

**Comment by livenplay** [comment]
Hi,
Firstly, @apasccon , thx for your work, it's great.
Secondly, I'm experiencing the same issue, so it's still there. 
This bug is hard to reproduce, but I was able to do it on the app from Example folder on this repo using Network Link Conditioner to make simulation of lower connectivity speed.
The main idea is to add some letter during downloading the results from the API, when activity indicator is shown. Normally, API calls are fast when you have a good Internet, thats why it's so hard to reproduce.
I've done a screen record from the simulator - https://drive.google.com/file/d/0B3YD-dq5pYydWEtkQWx5SVVwS0k
As you can mention, the filter items are set correctly, cause when I remove 1 letter the right result is visible, but it seems that the tableview frame isn't updating or probably the values are the same, so `layoutSubviews()` isn't called. 
I've tried to move `buildSearchTableView()` to filterDataSource observer instead of just redrawing the tableview in my app, (*I'm not using inline mode) and it worked well, but I'm not sure that it's the best solution.
 
```
fileprivate var filterDataSource = [SearchTextFieldItem]() {
      didSet {
            filter(false)
            //redrawSearchTableView()
            buildSearchTableView()
        }
 }
```
